boards/DIM/Src/Io/Io_SevenSegDisplays.c

```diff
@@ -0,0 +1,93 @@
+#include <stm32f3xx_hal.h>
+
+#include "App_SharedAssert.h"
+#include "App_SharedConstants.h"
+
+#include "Io_SevenSegDisplays.h"
+
+static_assert(HEX_DIGIT_0 == 0, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_1 == 1, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_2 == 2, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_3 == 3, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_4 == 4, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_5 == 5, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_6 == 6, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_7 == 7, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_8 == 8, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_9 == 9, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_A == 10, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_B == 11, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_C == 12, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_D == 13, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_E == 14, "Hex enum must match its numeric value.");
+static_assert(HEX_DIGIT_F == 15, "Hex enum must match its numeric value.");
+
+struct SevenSegDisplay_Commands
+{
+    uint8_t left;
+    uint8_t middle;
+    uint8_t right;
+};
+
+static SPI_HandleTypeDef *_hspi;
+
+static struct SevenSegDisplay_Commands commands;
+
+static const uint8_t command_lookup_table[NUM_HEX_DIGITS] = {
+    0x03, // 0x0
+    0x4F, // 0x1
+    0x25, // 0x2
+    0x0D, // 0x3
+    0x99, // 0x4
+    0x49, // 0x5
+    0x41, // 0x6
+    0x1F, // 0x7
+    0x01, // 0x8
+    0x09, // 0x9
+    0x11, // 0xA
+    0x1C, // 0xB
+    0x63, // 0xC
+    0x85, // 0xD
+    0x61, // 0xE
+    0x71, // 0xF
+};
```

Comment:
```suggestion
static const uint8_t command_lookup_table[NUM_HEX_DIGITS] = {
    0x3F, // 0x0
    0x06, // 0x1
    0x5B, // 0x2
    0x4F, // 0x3
    0x64, // 0x4
    0x6D, // 0x5
    0x7D, // 0x6
    0x07, // 0x7
    0x7F, // 0x8
    0x67, // 0x9
    0x77, // 0xA
    0x7C, // 0xB
    0x39, // 0xC
    0x5E, // 0xD
    0x79, // 0xE
    0x71, // 0xF
};
```

Reply:
Did we want to place this in a function?
```
uint8_t get_command_data(uint8_t hex_digit){
    static const uint8_t command_lookup_table[NUM_HEX_DIGITS] = {
        0x3F, // 0x0
        0x06, // 0x1
        0x5B, // 0x2
        0x4F, // 0x3
        0x64, // 0x4
        0x6D, // 0x5
        0x7D, // 0x6
        0x07, // 0x7
        0x7F, // 0x8
        0x67, // 0x9
        0x77, // 0xA
        0x7C, // 0xB
        0x39, // 0xC
        0x5E, // 0xD
        0x79, // 0xE
        0x71, // 0xF
        };
    return command_lookup_table[hex_digit];
}

Reply:
To set the decimal point the command (ie: 0x3F, // 0x0) would need to be or'd with 0x80. If we put the LUT in a getter function and then have a add decimal point boolean parameter we could or the LUT value with 0x80 at before returning if we want a decimal point?

Reply:
Don't feel very strongly about it because I don't see what value it adds at the moment and would rather keep it simple :thinking:

Reply:
Ya sure. Where would we want to enable the decimal point then?

Reply:
I was thinking about this too. Do we actually need to use the decimal point? I was reading the specs but didn't see an explicit need for it

Reply:
Ya we could get away without it, it would only be used for SoC. When cycling between displaying the SoC and speed, the decimal point on the SoC could help distinguishing between the two, but again it's not necessary.

Reply:
Let's cross that bridge when you get there and we can add decimal point is there a proven use-case